src/main/java/com/microsoft/sqlserver/jdbc/SQLServerCallableStatement.java

```diff
@@ -1451,6 +1451,12 @@ public NClob getNClob(String parameterName) throws SQLException {
         if (paramNames != null)
             l = paramNames.size();
 
+        // handle `@name` as well as `name`, since `@name` is what's returned 
+        // by DatabaseMetaData#getProcedureColumns
+        if (columnName.startsWith("@")) {
```

Comment:
we can add the check at this [line ](https://github.com/gordthompson/mssql-jdbc/blob/6789eb6fc86fe5b31a1ffd769f87065a616de2c2/src/main/java/com/microsoft/sqlserver/jdbc/SQLServerCallableStatement.java#L1472)
```
if (!columnName.startsWith("@"))
                sParam = sParam.substring(1, sParam.length());
```

The reason I am proposing this is that to have the original column name when we throw the exception at this [line ](https://github.com/gordthompson/mssql-jdbc/blob/6789eb6fc86fe5b31a1ffd769f87065a616de2c2/src/main/java/com/microsoft/sqlserver/jdbc/SQLServerCallableStatement.java#L1495) Otherwise looks great! thanks

Reply:
@v-afrafi - Good point. Your change would actually have to be made in two places, so I tweaked it a bit.